### PR TITLE
fix: unbound-method lint error by using arrow function syntax for fns declared on the ObservableQueryFields interface

### PR DESCRIPTION
**Pull request by alessbell**
Closes https://github.com/apollographql/apollo-client/issues/11554.

### CODE REVIEW

**Comment by alessbell** [comment]
Confirmed this would impact the docs. Any ideas on how we can make both happy, @phryneas? We could try @jerelmiller's idea of a "fake this" but that would also impact the type that ends up in the docs.

File: src/react/types/types.ts [hunk 1]
```diff
@@ -81,9 +81,11 @@ export interface ObservableQueryFields<
   TVariables extends OperationVariables,
 > {
   /** {@inheritDoc @apollo/client!QueryResultDocumentation#startPolling:member} */
-  startPolling(pollInterval: number): void;
+  // startPolling(pollInterval: number): void;
+  startPolling: ObservableQuery<TData, TVariables>["startPolling"];
```
Comment (phryneas) [inline src/react/types/types.ts:85]: Alternative options:

Arrow function
```suggestion
  startPolling: (pollInterval: number) => void;
```

`this: void,`
```suggestion
  startPolling(this: void, pollInterval: number): void;
```

**Comment by alessbell** [comment]
Thanks @phryneas, this is ready for a final look!

File: src/react/types/types.ts [hunk 2]
```diff
@@ -81,11 +81,13 @@ export interface ObservableQueryFields<
   TVariables extends OperationVariables,
 > {
   /** {@inheritDoc @apollo/client!QueryResultDocumentation#startPolling:member} */
-  startPolling(pollInterval: number): void;
+  // startPolling(pollInterval: number): void;
```
Comment (jerelmiller) [inline src/react/types/types.ts:84]:
```suggestion
```

I think we could probably get rid of the commented old code 🙂. What do you think?

File: src/react/types/types.ts [hunk 3]
```diff
@@ -81,11 +81,13 @@ export interface ObservableQueryFields<
...
+  startPolling: (pollInterval: number) => void;
   /** {@inheritDoc @apollo/client!QueryResultDocumentation#stopPolling:member} */
-  stopPolling(): void;
+  stopPolling: () => void;
+  // stopPolling(): void;
```
Comment (jerelmiller) [inline src/react/types/types.ts:88]:
```suggestion
```

**Review by jerelmiller — APPROVED** [review]
Looks great!

File: src/react/types/types.ts (hunk 2 above)
Comment (alessbell) [reply src/react/types/types.ts]: Oops, totally overlooked those :) Removed in https://github.com/apollographql/apollo-client/pull/11558/commits/556b085095f339cfc0cacbab89fdffbb56e1212e

**Review by jerelmiller — APPROVED** [review]
🚢 it

**Pull request merged**